packages/keychain/src/utils/account.ts

```diff
@@ -287,90 +303,102 @@ class Account extends BaseAccount {
 
     details.nonce = details.nonce ?? (await super.getNonce("latest"));
 
-    const overall_fee = number.toBN("10000000");
-    const fees: EstimateFee = {
-      overall_fee,
-      gas_consumed: number.toBN(0),
-      gas_price: number.toBN("100"),
-      suggestedMaxFee: overall_fee,
-    };
+    if (this.status === Status.PENDING_REGISTER) {
+      const pendingRegister = Storage.get(
+        selectors[VERSION].register(this.address, this._chainId),
+      );
+
+      const nextNonce = num.toHex(BigInt(details.nonce) + 1n);
+      const signerDetails: InvocationsSignerDetails = {
+        walletAddress: this.address,
+        nonce: nextNonce,
+        maxFee: constants.ZERO,
+        version: "0x2",
+        chainId: this._chainId,
+        cairoVersion: this.cairoVersion,
+      };
+
+      const signature = await this.signer.signTransaction(calls, signerDetails);
+      const invocation = {
+        contractAddress: this.address,
+        calldata: transaction.fromCallsToExecuteCalldata(calls),
+        signature,
+      } as Invocation;
+      const invokeDetails = {
+        nonce: nextNonce,
+        maxFee: constants.ZERO,
+        version: "0x2",
+      } as InvocationsDetailsWithNonce;
+      const invocationBulk = [
+        pendingRegister.invoke,
+        { invocation, invokeDetails },
+      ];
+
+      const estimates = await this.rpc.getEstimateFeeBulk(invocationBulk, {});
+
+      const fees = estimates.reduce<EstimateFee>(
+        (prev, estimate) => {
+          const overall_fee = prev.overall_fee + estimate.overall_fee;
+          return {
+            ...prev,
+            overall_fee: overall_fee,
+            gas_consumed: prev.gas_consumed + estimate.gas_consumed,
+            gas_price: prev.gas_price + estimate.gas_price,
+            suggestedMaxFee: overall_fee,
+            // TODO(#244): Set resource bounds
+            resourceBounds: {
+              l1_gas: {
+                max_amount: "",
+                max_price_per_unit: "",
+              },
+              l2_gas: {
+                max_amount: "",
+                max_price_per_unit: "",
+              },
```

Comment:
Not sure what to put. and why `overall_fee` and `gas_consumed` field still exist?